### PR TITLE
Hosting: Show success notification after Atomic transfer

### PR DESCRIPTION
**Pull request by mmtr**
Fixes https://github.com/Automattic/dotcom-forge/issues/9212

## Proposed Changes

Displays a success notification after activating the hosting features:

Screenshot 2024-09-26 at 12 29 42

## Why are these changes being made?

To inform the user that their hosting features have been activated.

## Testing Instructions

- Use the Calypso live link below
- Go to `/sites`
- Select a Simple site with a Business plan
- Open the "Hosting Features" tab
- Activate them
- Wait for the Atomic transfer to finish
- Make sure a success notification shows up

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/hosting/overview/controller.tsx [hunk 1]
```diff
@@ -1,14 +1,28 @@
...
+	if ( context.query && context.query.hosting_features === 'activated' ) {
+		context.store.dispatch(
+			successNotice( i18n.translate( 'Hosting features activated successfully!' ), {
+				displayOnNextPage: true,
+			} )
+		);
+		// Remove query param without triggering a re-render
+		const newUrl = new URL( window.location.href );
+		newUrl.searchParams.delete( 'hosting_features' );
+		window.history.replaceState( null, '', newUrl.toString() );
```
Comment (xavier-lc) [inline client/hosting/overview/controller.tsx]:
```suggestion
		window.history.replaceState( null, '', removeQueryArgs( window.location.href, 'hosting_features' ) );
```

I see we have this approach in other places, it looks a little bit clearer:

https://github.com/Automattic/wp-calypso/blob/7e593ecd88c0ce4b4a2bedcd59a95c9bdb53a83d/client/jetpack-cloud/sections/agency-dashboard/sites-overview/index.tsx#L299-L304

**Review by xavier-lc — APPROVED** [review]
Works as described 👌

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/16889020

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @mmtr for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/16889020).

**Pull request merged**